includes/admin/class-wp-job-manager-promoted-jobs-admin.php

```diff
@@ -260,35 +260,43 @@ public function promoted_jobs_custom_columns( $column ) {
 	 * @return string
 	 */
 	public function get_promote_jobs_template() {
-		$promote_template_option_name          = 'promote-jobs-template_' . get_user_locale();
-		$promote_jobs_template_next_check_name = '_promote-jobs-template_next_check_' . get_user_locale();
-		$promote_template                      = get_option( $promote_template_option_name, false );
-		$promote_jobs_template_next_check      = get_option( $promote_jobs_template_next_check_name );
-		$check_for_updated_template            = false;
-
-		if ( ! $promote_jobs_template_next_check || $promote_jobs_template_next_check < time() ) {
-			$check_for_updated_template = true;
+		$locale                          = get_user_locale();
+		$promote_template_transient_name = 'jm_promote-jobs-template_' . $locale;
+		$promote_template                = get_transient( $promote_template_transient_name );
+
+		if ( false !== $promote_template ) {
+			return $promote_template;
 		}
 
-		if ( $check_for_updated_template ) {
-			$response = wp_safe_remote_get( 'https://wpjobmanager.com/wp-json/promoted-jobs/v1/assets/promote-dialog/?lang=' . get_user_locale() );
-			if (
-				is_wp_error( $response )
-				|| 200 !== wp_remote_retrieve_response_code( $response )
-				|| empty( wp_remote_retrieve_body( $response ) )
-			) {
-				update_option( $promote_jobs_template_next_check_name, time() + MINUTE_IN_SECONDS * 5, false );
-				return $promote_template;
-			} else {
-				$assets           = json_decode( wp_remote_retrieve_body( $response ), true );
-				$promote_template = $assets['assets'][0]['content'];
-				update_option( $promote_template_option_name, $promote_template, false );
-				update_option( $promote_jobs_template_next_check_name, time() + HOUR_IN_SECONDS, false );
-			}
+		$url      = 'https://wpjobmanager.com/wp-json/promoted-jobs/v1/assets/promote-dialog/?lang=' . $locale;
+		$response = wp_safe_remote_get( $url );
+		$fallback = '
+			<div>
+				<br />
```

Comment:
I started trying with a message, but it was a little ugly. So I realized that maybe it wasn't very important because we're presenting in the modal what the user really needs. The information they are missing in the modal we will probably also have on the "Learn more" page.

But if you have a strong opinion on it, I could write some CSS to customize it and make it not so ugly with a message.

Reply:
By "presenting in the modal what the user really needs" you mean having the learn more and promote button working, right?

If yes, it is fine, although a little weird. :thinking: I'll consider this a non-issue.

Reply:
Correct! I'll keep it as it is for now then, and anything we could iterate later. ;)